src/sequelize-typescript.ts

```diff
@@ -170,4 +175,148 @@ export class SequelizeTypeScript {
+   *
+   * ```ts
+   * const sequelize = new Sequelize({
+   *   // ...
+   *   disableAlsTransactions: true,
+   * })
+   *
+   * await sequelize.transaction(transaction => {
+   *   // transactions are not automatically passed around anymore, you need to do it yourself:
+   *   const user = await User.findOne(..., { transaction });
+   *   await user.update(..., { transaction });
+   * });
+   * ```
+   *
+   * If you want to manage your transaction yourself, use {@link startUnmanagedTransaction}.
+   *
+   * @param callback Async callback during which the transaction will be active
+   */
+  transaction<T>(callback: TransactionCallback<T>): Promise<T>;
+  /**
+   * @param options Transaction Options
+   * @param callback Async callback during which the transaction will be active
+   */
+  transaction<T>(options: TransactionOptions, callback: TransactionCallback<T>): Promise<T>;
+  async transaction<T>(
+    optionsOrCallback: TransactionOptions | TransactionCallback<T>,
+    maybeCallback?: TransactionCallback<T>,
+  ): Promise<T> {
+    let options: TransactionOptions;
+    let callback: TransactionCallback<T>;
+    if (typeof optionsOrCallback === 'function') {
+      callback = optionsOrCallback;
+      options = {};
+    } else {
+      callback = maybeCallback!;
+      options = optionsOrCallback;
+    }
+
+    if (!callback) {
+      throw new Error('sequelize.transaction requires a callback. If you wish to start an unmanaged transaction, please use sequelize.startUnmanagedTransaction instead');
+    }
+
+    const transaction = new Transaction(
+      // @ts-expect-error -- remove once this class has been merged back with the Sequelize class
+      this,
+      options,
+    );
+
+    const wrappedCallback = async () => {
+      await transaction.prepareEnvironment();
+
+      let result;
+      try {
+        result = await callback(transaction);
+      } catch (error) {
+        try {
+          await transaction.rollback();
+        } catch {
+          // ignore, because 'rollback' will already print the error before killing the connection
+        }
+
+        throw error;
+      }
+
+      await transaction.commit();
+
+      return result;
+    };
+
+    const als = this.#transactionAls;
+    if (!als) {
+      return wrappedCallback();
+    }
+
+    return als.run(transaction, wrappedCallback);
+  }
+
+  /**
+   * We highly recommend using {@link Sequelize#transaction} instead.
+   * If you really want to use the manual solution, don't forget to commit or rollback your transaction once you are done with it.
+   *
+   * Transactions started by this method are not automatically passed to queries. You must pass the transaction object manually,
+   * even if {@link Options.disableAlsTransactions} is false.
+   *
+   * @example
+   * ```ts
+   * try {
+   *   const transaction = await sequelize.unsafeStartManualTransaction();
```

Comment:
```suggestion
   *   const transaction = await sequelize.startUnmanagedTransaction();
```

Reply:
Thank you for reviewing :) Done

Reply:
I was just curious what's going to happen to unmaintained `cls-hooked` package and managed transactions in Sequelize, and went across this PR😅